### DISCUSSION TITLE
Custom NFT, should I use CID on Interactions.s.sol?

### DISCUSSION THREAD

**Discussion by yawarasuuna** [Cyfrin/foundry-full-course-cu#2862] [Q&A] [answered]
I tried to do a custom NFT while following along Advanced Foundry, Section 2, Lesson 9, until [around 2m29s](https://updraft.cyfrin.io/courses/advanced-foundry/how-to-create-an-NFT-collection/interact-with-solidity-smart-contracts), but once a deployed my contract on Sepolia, the image is not showing up

I tried a few solutions suggested in other discussion, like changing metamask settings, but it does not show up, so I decided to just replace my custom ipfs link with the PUG version on the Interactions.s.sol , and everything works now and the PUG is showing with it:

```
"ipfs://bafybeig37ioir76s7mg5oobetncojcm3c3hxasyd4rvid4jqhy4gkaheg4/?filename=0-PUG.json";
```

```
// SPDX-License-Identifier: MIT

pragme solidity ^0.8.18;

import {Script} from "forge-std/Script.sol";
import {BasicNft} from "../src/BasicNft.sol";
import {DevOpsTools} from "lib/foundry-devops/src/DevOpsTools.sol";

contract MintBasicNft is Script{

    string public constant TOKENURI =
        "ipfs://bafybeig37ioir76s7mg5oobetncojcm3c3hxasyd4rvid4jqhy4gkaheg4/?filename=0-PUG.json";

    function run() external {
        address mostRecentlyDeployed = DevOpsTools.get_most_recent_deployment("BasicNft", block.chainid);

        mintNftOnContract(mostRecentlyDeployed);
    }

    function mintNftOnContract(address contractAddress) public {
        vm.startBroadcast();
        BasicNft(contractAddress).mintNft(TOKENURI)
        vm.stopBroadcast();
    }
}
```

My custom JSON is pointing to a link "ipfs://CIDHere", instead of "https://ipfs.io/ipfs/CIDHere"
When I deployed my contract, I used the "share link" from the IPFS desktop app, seen that it was the only one that had an ending similar to the PUG with a ".json" in the end, but it is not exactly the same:

```
https://bafyxfgRandomStringthersgrthrthdszvbne4riougn4f3aewoi8gn4**.ipfs.dweb.link**?filename=Fiddle1.json
```

Perhaps this might be the mistake I'm making?
I also tried using only the CID and CID+.json:

```
 "ipfs://bafyxfgRandomStringthersgrthrthdszvbne4riougn4f3aewoi8gn4";
 "ipfs://bafyxfgRandomStringthersgrthrthdszvbne4riougn4f3aewoi8gn4.json";
```

 Or is my json incorrect? See below:

```
{
    "name": "Bonish Fiddle",
    "description": "Fiddling with fiddles",
    "image": "ipfs://ImageCIDGoesHere",
    "attributes": [
        {
            "trait_type": "bony",
            "value": 100
        }
    ]
}
```

Or perhaps my ipfs files are only available locally ? I have no clue now lol

Thanks in advance for any help

**Comment by EngrPips** [comment, marked as answer]
Yeah, your ipfs files are only readily served when your ipfs node is running, and the moment you shut down your node, there won't be any way for any node to fetch your files anymore except if some other node finds your file useful and pin it. But you can host your file to some node service that is always up and running, and that way, your files can always be available.
